pkg/cluster/cluster.go

```diff
@@ -162,7 +152,12 @@ func (c *Cluster) setup() error {
 		return errCreatedCluster
 	case spec.ClusterPhaseRunning:
 		shouldCreateCluster = false
-
+	case spec.ClusterPhaseFailed:
```

Comment:
I suggest we change phase to running here and set shouldCreateCluster to false and continue instead of panic

Reply:
Sounds good, fixed

Reply:
Applied on a Failed cluster, it works.